starter-code/app.js

```diff
@@ -1,5 +1,75 @@
+/*jshint esversion: 6 */
 const express = require('express');
 const app = express();
 const Chuck  = require('chucknorris-io');
 const client = new Chuck();
+const bodyParser = require('body-parser');
 
+
+app.use(bodyParser.urlencoded({ extended: true }));
+
+app.set('views', __dirname + '/views');
+app.set('view engine', 'ejs');
+
+app.get('/random', (req, res, next) => {
+
+  client.getRandomJoke()
+  .then((response) => {
+    // use the esponse here
+    console.log(response);
+    res.send( `<p>${response.value}</p>`);
+  }).catch((err) => {
+    // handle error
+  });
+});
+
+app.get('/categories', (req, res, next) => {
+if(req.query.cat === undefined){
+  client.getJokeCategories()
+    .then((response)=>  {
+      // use the response here
+      res.render("categories",{categories:response});
```

Comment:
You can write it shorter by changing response to categories: 

 client.getJokeCategories().then(function (categories) {
    res.render('categories', {categories});